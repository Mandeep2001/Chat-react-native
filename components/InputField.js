import React, { useState } from 'react';
import { StyleSheet, TextInput } from 'react-native';
import PropTypes from 'prop-types';

const styles = StyleSheet.create({});

const InputField = ({ placeHolder, name, onChangeHandler }) => {
  const [value, setValue] = useState('');

  return (
    <TextInput
      value={value}
      placeholder={placeHolder}
      onChange={text => {
        setValue(text);
        onChangeHandler(name, value);
      }}
    />
  );
};

InputField.propTypes = {
  placeHolder: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  onChangeHandler: PropTypes.func.isRequired,
};

export default InputField;
