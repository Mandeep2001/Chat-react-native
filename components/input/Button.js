import React from 'react';
import { TouchableOpacity, Text, ViewPropTypes, StyleSheet } from 'react-native';
import PropTypes from 'prop-types';
import { Colors, Fonts } from '../../style/styles';

const styles = StyleSheet.create({
  button: { borderRadius: 8, backgroundColor: Colors.primary, alignItems: 'center' },
  text: { ...Fonts.body, color: '#fff', paddingVertical: 10 },
});

const PrimaryButton = ({ text, pressHandler, style }) => {
  return (
    <TouchableOpacity onPress={pressHandler} style={{ ...styles.button, ...style }}>
      <Text style={styles.text}>{text}</Text>
    </TouchableOpacity>
  );
};

PrimaryButton.propTypes = {
  text: PropTypes.string.isRequired,
  pressHandler: PropTypes.func.isRequired,
  style: ViewPropTypes.style,
};

PrimaryButton.defaultProps = { style: {} };

export default PrimaryButton;
