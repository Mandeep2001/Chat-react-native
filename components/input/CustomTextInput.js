import React from 'react';
import { View, TextInput } from 'react-native';
import PropTypes from 'prop-types';

const CustomTextInput = ({ value, onChangeText, style, placeholder }) => {
  return (
    <View>
      <TextInput
        placeholder={placeholder}
        value={value}
        onChangeText={onChangeText}
        style={style}
      />
    </View>
  );
};

CustomTextInput.propTypes = {
  value: PropTypes.string.isRequired,
  onChangeText: PropTypes.func.isRequired,
  style: PropTypes.instanceOf(Object).isRequired,
  placeholder: PropTypes.string.isRequired,
};

export default CustomTextInput;
