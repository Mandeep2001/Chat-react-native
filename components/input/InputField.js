import React from 'react';
import { StyleSheet, TextInput, ViewPropTypes } from 'react-native';
import PropTypes from 'prop-types';
import { Colors } from '../../style/styles';

const styles = StyleSheet.create({
  textInput: {
    borderColor: Colors.lightGray,
    borderWidth: 0.5,
    backgroundColor: Colors.lowConstrastGray,
    borderRadius: 8,
    paddingHorizontal: 16,
    paddingVertical: 8,
    fontSize: 17,
  },
  textInputError: {
    borderColor: Colors.error,
    borderWidth: 0.5,
    backgroundColor: Colors.errorBackground,
    color: Colors.error,
    borderRadius: 8,
    paddingHorizontal: 16,
    paddingVertical: 8,
    fontSize: 17,
  },
});

const InputField = ({ placeHolder, onChangeHandler, style, secureTextEntry, error, value }) => {
  return (
    <TextInput
      style={error ? { ...styles.textInputError, ...style } : { ...styles.textInput, ...style }}
      placeholder={placeHolder}
      value={value}
      placeholderTextColor={error ? Colors.error : '#C7C7CD'}
      onChangeText={text => onChangeHandler(text)}
      secureTextEntry={secureTextEntry}
    />
  );
};

InputField.propTypes = {
  placeHolder: PropTypes.string.isRequired,
  onChangeHandler: PropTypes.func.isRequired,
  style: ViewPropTypes.style,
  secureTextEntry: PropTypes.bool,
  error: PropTypes.bool.isRequired,
  value: PropTypes.string.isRequired,
};

InputField.defaultProps = {
  style: {},
  secureTextEntry: false,
};

export default InputField;
