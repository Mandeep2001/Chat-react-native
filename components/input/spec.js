/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { shallow } from 'enzyme';
import Button from './Button';
import CustomTextInput from './CustomTextInput';
import InputField from './InputField';

describe('Button', () => {
  describe('Rendering', () => {
    it('should match to snapshot', () => {
      const component = shallow(
        <Button pressHandler={() => console.log('Premuto')} text="Button" />
      );
      expect(component).toMatchSnapshot();
    });
  });
});

describe('CustomTextInput', () => {
  describe('Rendering', () => {
    it('should match to snapshot', () => {
      let value = 'initialValue';
      const onChangeText = v => {
        value += v;
      };

      const component = shallow(
        <CustomTextInput
          placeholder="CustomTextInput"
          value={value}
          onChangeText={onChangeText}
          style={{}}
        />
      );
      expect(component).toMatchSnapshot();
    });
  });
});

describe('InputField', () => {
  describe('Rendering', () => {
    it('should match to snapshot', () => {
      let value = 'initialValue';
      const onChangeText = v => {
        value += v;
      };

      const component = shallow(
        <InputField
          placeHolder="Input field"
          onChangeHandler={onChangeText}
          value={value}
          error={false}
          secureTextEntry
        />
      );

      expect(component).toMatchSnapshot();
    });
  });
});
