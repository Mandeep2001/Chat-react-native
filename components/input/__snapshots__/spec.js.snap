// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Rendering should match to snapshot 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#33E1ED",
      "borderRadius": 8,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#fff",
        "fontSize": 17,
        "fontWeight": "normal",
        "paddingVertical": 10,
      }
    }
  >
    Button
  </Text>
</TouchableOpacity>
`;

exports[`CustomTextInput Rendering should match to snapshot 1`] = `
<View>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="CustomTextInput"
    placeholderTextColor={null}
    rejectResponderTermination={true}
    style={Object {}}
    underlineColorAndroid="transparent"
    value="initialValue"
  />
</View>
`;

exports[`InputField Rendering should match to snapshot 1`] = `
<TextInput
  allowFontScaling={true}
  onChangeText={[Function]}
  placeholder="Input field"
  placeholderTextColor="#C7C7CD"
  rejectResponderTermination={true}
  secureTextEntry={true}
  style={
    Object {
      "backgroundColor": "#F3F3F3",
      "borderColor": "#707070",
      "borderRadius": 8,
      "borderWidth": 0.5,
      "fontSize": 17,
      "paddingHorizontal": 16,
      "paddingVertical": 8,
    }
  }
  underlineColorAndroid="transparent"
  value="initialValue"
/>
`;
